### PR TITLE
lua: SHA3 implementation

### LINKED ISSUE #1655: SHA-3 Implementation
Hi Tarantool team,
   Great job with the database - I am looking forward to trying it out!  

One request: is there any plan to add the SHA-3 algorithm to crypto module?  There is an implementation available for SHA-3 in Lua mailing group.

### PR DESCRIPTION
**Pull request by ortho2005**
Fixes #1655 

Add bindings sha3 from openssl 

```
digest = require('digest')

digest.sha3_224()
digest.sha3_256()
digest.sha3_384()
digest.sha3_512()
```
also
```
crypto = require('crypto')

crypto.hmac.sha3_224()
crypto.hmac.sha3_256()
crypto.hmac.sha3_384()
crypto.hmac.sha3_512()
```

### CODE REVIEW

**Comment by ortho2005** [comment]
Test failed because of this
```
openssl version
76OpenSSL 1.0.2g  1 Mar 2016
```

What shall I do? 
For example I can remove test for new SHA3 method.

**Comment by Totktonada** [comment]
I would expose openssl version information from `tarantool --version` output (and from built-in `tarantool` module), extract SHA3 test cases into a separate test file and add .skipcond file to skip the test on old openssl versions.

However if you'll just accept `'HMAC method "sha3.*" is not supported'` error for those cases I would not mind. It is a bit tricky for tests, which compares test output with a reference, so I suggest you to consider to add a separate TAP13 test into the `test/app-tap` directory.

**Comment by ortho2005** [comment]
OK. The main way idea is clear for me.  I'll try to do something with tests

**Comment by ortho2005** [comment]
I giving up writing skipcond file for tests. Sorry :(

All I've done is extract test to differents files and force them to skip. 
There is also new function crtypto_ssl_version wrapper around OpenSSL_version 
Tarantool module give access to it. 
That's all

File: src/lua/crypto.lua [hunk 1]
```diff
@@ -81,7 +81,7 @@ for class, name in pairs({
     sha256 = 'SHA256', sha384 = 'SHA384', sha512 = 'SHA512',
     sha3_224 = 'SHA3-224', sha3_256 = 'SHA3-256', sha3_384 = 'SHA3-384', sha3_512 = 'SHA3-512',
     dss = 'DSS', dss1 = 'DSS1', mdc2 = 'MDC2', ripemd160 = 'RIPEMD160'}) do
-    local digest = ffi.C.EVP_get_digestbyname(class)
+    local digest = ffi.C.EVP_get_digestbyname(name)
```
Comment (romanhabibov) [inline src/lua/crypto.lua:84]: Add `name` in `for` condition, now it is named as `_`.

**Review by romanhabibov — COMMENTED** [review]
Please, squash `split sha3 function to different test files`, `Add test for new sha3_* functions` and `Register SHA3 in lua` into one commit.

**Comment by romanhabibov** [comment]
Please, checkout all commits according to https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message

**Review by romanhabibov — COMMENTED** [review]
IMO, you should drop `wrap OpenSSL_version function`, `add ssl_version function to tarantool module` and make `.skipcond` file based on `popen` + `openssl version` combination. Why introduce `tarantool --version` for the sake of a single test, when you can get by with the method I suggested?

**Comment by kyukhin** [comment]
No reaction for year+.

**Pull request closed without merging**